Traffic weighting

mark-church opened the pull request:
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This adds a documentation guide with realistic examples for traffic shifting.

Question for reviewers: There are many words for this - traffic shifting, traffic splitting, traffic weighting. Is there a preference that we have in this project?

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes-sigs/service-apis/issues/510

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

robscott commented:
Thanks!

/ok-to-test

robscott commented on docs-src/traffic-shifting.md:
```diff
@@ -0,0 +1,97 @@
+# HTTP traffic shifting
```
Tiny Nit: Is "Traffic Splitting" or "Traffic Shifting" a more common term? It looks like both terms are used throughout the doc, just trying to think of which users will be more likely to search for.

robscott replied:
Just saw that you asked about this in the PR as well. Between traffic shifting, traffic splitting, and traffic weighting, I have a slight preference towards traffic splitting, followed by traffic shifting. Interested in what others think as well.

hbagdi replied:
I like traffic splitting over shifting.

mark-church replied:
traffic splitting wins then! I'm switching over everything for traffic splitting.

robscott commented on docs-src/traffic-shifting.md:
```diff
@@ -0,0 +1,97 @@
+    - serviceName: foo-svc-2
+      port: 8080
+      weight: 10
+...
```
What do the dots here represent? I'm probably forgetting something obvious, but would this be a valid route as is?

hbagdi replied:
I think it is best to `include` a valid HTTPRoute resource here. It is verbose but allows users to copy-paste and hit the ground running.

mark-church replied:
okay I added in the full resource with an include

robscott commented on docs-src/traffic-shifting.md:
```diff
@@ -0,0 +1,97 @@
+## Canary traffic
+
+At first, there may only be a single version of a Service that serves
+production user  traffic for `foo.example.com`. The following HTTPRoute has no
```
Tiny nit:
```suggestion
production user traffic for `foo.example.com`. The following HTTPRoute has no
```

robscott commented on examples/traffic-shifting/foo-traffic-shift-1.yaml:
```diff
@@ -0,0 +1,21 @@
+          traffic: test
+    forwardTo:
+    - serviceName: foo-svc-v2
+      port: 8080
```
Nit: need a new line

robscott commented on examples/traffic-shifting/foo-traffic-shift-2.yaml:
```diff
@@ -0,0 +1,17 @@
+      weight: 90
+    - serviceName: foo-svc-v2
+      port: 8080
+      weight: 10
```
Nit: need a new line

robscott commented on examples/traffic-shifting/foo-traffic-shift-3.yaml:
```diff
@@ -0,0 +1,17 @@
+      weight: 0
+    - serviceName: foo-svc-v2
+      port: 8080
+      weight: 1
```
Nit: need a new line

robscott commented on examples/traffic-shifting/gateway.yaml:
```diff
@@ -0,0 +1,14 @@
+      kind: HTTPRoute
+      selector:
+        matchLabels:
+          gateway: prod-web-gw
```
Nit: need a new line

robscott reviewed (commented):
Thanks @mark-church! This will be a great addition to our docs! Mostly LGTM, just had a few nits

hbagdi commented:
Other than comment from Rob, this looks nearly ready for merge.
@mark-church Thanks for putting this together, this is good stuff!

mark-church commented:
Latest changes ....

- Changed everything from "traffic shifting" to "traffic splitting"
- Use only includes from examples folder
- Updated and simplified examples

robscott commented:
Thanks!

/lgtm
/approve
